app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,38 @@
+class CommentsController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @comments = Comment.all
+  end
+
+def create
+   @comment = Comment.new(comment_params)
+ if @comment.save
+    @post = @comment.post
+     redirect_to @post 
+    flash.notice = "Comment added"
+   else
+    redirect_to root_path
+   flash.alert = "Invalid comment"
+  end
+end
+
+def destroy
+  @comment = Comment.find_by(id: params[:id])
+  if @comment.destroy
+    flash.alert = "Comment deleted"
+    redirect_to request.referrer 
```

Comment:
no matter what we will redirect, let's do it once at the end of everything.

Reply:
I don't understand this. Could you explain a little bit more?